services/movie.py

```diff
@@ -0,0 +1,50 @@
+from django.db.models import QuerySet
+
+from db.models import Movie
+
+
+def get_movies(
+        genres_ids: list[int] = None,
+        actors_ids: list[int] = None
+) -> QuerySet:
+    queryset = Movie.objects.all()
+
+    if genres_ids and actors_ids:
+        queryset = queryset.filter(
+            genres__id__in=genres_ids,
+            actors__id__in=actors_ids
+        )
+
+    if genres_ids and not actors_ids:
+        queryset = queryset.filter(
+            genres__id__in=genres_ids
+        )
+
+    if actors_ids and not genres_ids:
+        queryset = queryset.filter(
+            actors__id__in=actors_ids
+        )
+
+    return queryset
```

Comment:
You can simplify the get_movies function by first applying one set of criteria (either genres or actors) and then applying the other set of criteria if available. 

First filter the queryset based on genres_ids if it's provided.
Then,  filter the already filtered queryset based on actors_ids if it's provided.